lib/active_model_serializers/model.rb

```diff
@@ -37,7 +51,7 @@ def initialize(attributes = {})
     # +attributes+ are returned frozen to prevent any expectations that mutation affects
     # the actual values in the model.
     def attributes
-      attribute_names.each_with_object({}) do |attribute_name, result|
+      self.class.attribute_names.each_with_object({}) do |attribute_name, result|
```

Comment:
removed the instance reader/writer since the ActiveModel::Serialization mixin in Rails uses an `attribute_names` local variable, which I found because it conflicted in some scenario I've forgotten